### PR TITLE
Data cleanup fixes and improvements

### PR DESCRIPTION
**Pull request by cjellick**

### CODE REVIEW

File: code/iaas/model/src/main/java/io/cattle/platform/core/cleanup/TableCleanup.java [hunk 1]
```diff
@@ -249,81 +269,115 @@ private void cleanupLabelTables(Date cutoff) {
...
+            }
 
-                    if ((toFilter = filterIds.fetch()).size() > 0) {
-                        for (Record1<Long> record : toFilter) {
-                            idsToFilter.add(record.value1());
-                        }
-                    }
+            // Build a set of ids referenced by other tables
+            List<ForeignKey<?, ?>> keys = getReferencesFrom(table, tables);
+            for (ForeignKey<?, ?> key : keys) {
```
Comment (alena1108) [inline code/iaas/model/src/main/java/io/cattle/platform/core/cleanup/TableCleanup.java:325]: Just curious - why do we iterate over the same set of keys twice? here an on line 330. Also is the intent to just log the error when composite key is present?

File: code/iaas/model/src/main/java/io/cattle/platform/core/cleanup/TableCleanup.java [hunk 2]
```diff
@@ -249,81 +269,115 @@ private void cleanupLabelTables(Date cutoff) {
...
+                if (idsToDelete.size() > 0) {
+                    try {
+                        int rowsDeleted = create()
+                                .delete(table.table)
+                                .where(idField.in(idsToDelete))
+                                .execute();
+                        table.addRowsDeleted(rowsDeleted);
+                        log.debug("Deleted {} unneeded rows from {} table", idsToDelete.size(), table.table);
+                    } catch (DataAccessException e) {
+                        log.info("Problem executing cleanup query: " + e.getMessage());
```
Comment (alena1108) [inline code/iaas/model/src/main/java/io/cattle/platform/core/cleanup/TableCleanup.java:365]: Nit: should we log it in WARN?

Comment (cjellick) [reply code/iaas/model/src/main/java/io/cattle/platform/core/cleanup/TableCleanup.java:365]: This was set to info before, i dont want to change the level

File: code/iaas/model/src/main/java/io/cattle/platform/core/cleanup/TableCleanup.java (hunk 1 above)
Comment (cjellick) [reply code/iaas/model/src/main/java/io/cattle/platform/core/cleanup/TableCleanup.java:325]: Pre-existing logic. I felt it best to leave as-is

**Comment by alena1108** [comment]
LGTM

**Comment by deniseschannon** [comment]
https://github.com/rancher/rancher/issues/10841

**Comment by deniseschannon** [comment]
https://github.com/rancher/rancher/issues/7050

**Pull request merged**